### ISSUE TITLE
Issue with matching

### ISSUE THREAD

**Issue by sethen** [nvim-tree/nvim-web-devicons#483]
Looks like nvim-web-devicons gets confused with you do something like this.  I would expect it to know that it should apply the same icon to `.ts` files as it does for `test.ts` files, but it doesn't.  To make it work I have to do something like this:

```lua
return {
   'nvim-tree/nvim-web-devicons',
   dependencies = {
      'catppuccin/nvim',
   },
   config = function()
      local icons = require('nvim-web-devicons')
      local palette = require('catppuccin.palettes').get_palette('mocha')

      icons.setup({
         override = {
            ts = {
               color = palette.blue,
               icon = '',
               name = 'Ts',
            },
            ['test.ts'] = {
               color = palette.blue,
               icon = '',
               name = 'Test.ts',
            },
         }
      })
      icons.set_default_icon('')
   end,
}
```

![Screenshot from 2024-07-21 00-07-02](https://github.com/user-attachments/assets/1933d100-3c06-4fb2-8f06-ca2b4340a859)

**Comment by hasecilu** [comment]
Are you sure it's a glyph from NF? Seems you're using a faulty codepoint.

**Comment by sethen** [comment]
> Are you sure it's a glyph from NF? Seems you're using a faulty codepoint.

Guess I don't understand how that would be a faulty codepoint?  How does it work with `.ts` extensions but when `test.ts` is in the file it doesn't work?

**Comment by hasecilu** [comment]
I meant that in your snippet you used codepoint EA55 and in NerdFonts there is no associated glyph. If you got a "square" there it's because there is no glyph there.

I tried replacing the icons and works:

![image](https://github.com/user-attachments/assets/83182891-1ba3-4133-b0aa-70847708305a)

**Comment by sethen** [comment]
> I meant that in your snippet you used codepoint EA55 and in NerdFonts there is no associated glyph. If you got a "square" there it's because there is no glyph there.
> 
> I tried replacing the icons and works:
> 
> ![image](https://private-user-images.githubusercontent.com/53124818/362414750-83182891-1ba3-4133-b0aa-70847708305a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ4NzE0ODksIm5iZiI6MTcyNDg3MTE4OSwicGF0aCI6Ii81MzEyNDgxOC8zNjI0MTQ3NTAtODMxODI4OTEtMWJhMy00MTMzLWIwYWEtNzA4NDc3MDgzMDVhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI4VDE4NTMwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNjNWI3MjFkNDU1M2ZiMWQyMzliZjIwMTAwOTQ1YWE3NTllYWNmZWUxYTNjMWI0Njg2OGE1NTE3M2VhY2U0NjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.eoH8bfJDWjTuULvoDYsVeVD-l3EbSKPI5CFDYySKPY4)

You're correct, there is no code point in NerdFonts.  I use my own custom icon font and define the codepoints in Kitty (Terminal).

**Comment by sethen** [comment]
What you're seeing with the code point (EA55) is the same code point for the `.ts` files.  So I am confused why it won't load the same icon for the same file.

**Comment by hasecilu** [comment]
I tried changing both extensions to the same char and also works. Not sure what could be happening on your end, maybe using `override_by_extension` makes a change.

**Comment by gegoune** [comment]
@sethen Is that still an issue on your end?

**Comment by alex-courtis** [comment]
> Okay so after some digging I found this: https://github.com/nvim-tree/nvim-web-devicons/blob/master/lua/nvim-web-devicons/light/icons_by_filename.lua#L167 -- how do I override these and tell devicons _not_ use this rule for this file??

You can override for the individual files by passing something like this to setup:
```lua
  override_by_filename = {
    ["pnpm-lock.yaml"] = {
      icon = "",
      color = "#6d8086",
      name = "PNPMLockOverride"
    }
  }
```

You might find `:NvimWebDeviconsHiTest` interesting. It will show you all icons that are configured and why e.g.

<img width="853" height="572" alt="Image" src="https://github.com/user-attachments/assets/0ff34ac1-27ec-464d-a492-2e71ad826392" />
